### ISSUE TITLE
Request to add function: "sys_extract_utc"

### ISSUE THREAD

**Issue by casparsch** [orafce/orafce#271]
Oracle currently provides the function: [SYS_EXTRACT_UTC](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/SYS_EXTRACT_UTC.html)
I think this can be adopted in the orafce extension by following postgres' counterpart:
```plpgsql
CREATE OR REPLACE FUNCTION sys_extract_utc(
	a_timestamp_with_timezone timestamp with time zone)
    RETURNS timestamp without time zone
    LANGUAGE 'sql'
    COST 10
    IMMUTABLE
    SECURITY DEFINER
    PARALLEL SAFE 
AS
$BODY$
    SELECT  a_timestamp_with_timezone AT TIME ZONE 'UTC';
$BODY$;
``` 
The properties of the function like COST, IMMUTABLE, PARALLEL, SEC-DEF are **my** choices. It adds no functionality and should/could be considered when adopted in orafce.

Knowing currently in UTC+02:00, tested it with:
```
SELECT  '2024-07-30 10:29:28.276325+02'::timestamptz
,       '2024-07-30 10:29:28.276325'::timestamptz
,       SYS_EXTRACT_UTC('2024-07-30 10:29:28.276325+02'::timestamptz)
,       SYS_EXTRACT_UTC('2024-07-30 10:29:28.276325'::timestamptz)
;
```

**Comment by okbob** [comment]
good idea, but I need to support oracle.date (it is timestamp(0)) probably too. I have not idea now, what should be correct behaviour of proposed function for oracle.date when orafce.timezone will be different than default "GMT".

**Comment by casparsch** [comment]
Found out following:
On Oracle parameter of datatype: "DATE" is **not** allowed for function: "SYS_EXTRACT_UTC"
So that should not be a problem. _Or am I not getting your concern, a misunderstanding?_

Therefore only timestamps (with or without timezone) was than my focus.
Made a sql script shown below with:
> - _first 8 lines are for formatting in sqlplus for a wide enough window (> 130 characters)_
> - _next 3 "alter session" statements to have a format that I prefer (though the first one isn't used)_
```
set linesize 130
set pagesize 100
column ts         format a40
column ts_tz      format a40
column ts_tz1     format a40
column utc_ts     format a40
column utc_ts_tz  format a40
column utc_ts_tz1 format a40

alter session set nls_date_format         = 'yyyy-mm-dd hh24:mi:ss';
alter session set nls_timestamp_format    = 'yyyy-mm-dd hh24:mi:ss.ff';
alter session set nls_timestamp_tz_format = 'yyyy-mm-dd hh24:mi:ss.ff tzh:tzm';

SELECT to_timestamp   ('2024-07-30 10:29:28.45678'                       , 'yyyy-mm-dd hh24:mi:ss.ff')          ts
,      to_timestamp_tz('2024-07-30 10:29:28.45678'                       , 'yyyy-mm-dd hh24:mi:ss.ff')          ts_tz
,      to_timestamp_tz('2024-07-30 10:29:28.45678 +01:00'                , 'yyyy-mm-dd hh24:mi:ss.ff TZH:TZM')  ts_tz1
,      SYS_EXTRACT_UTC(to_timestamp   ('2024-07-30 10:29:28.45678'       , 'yyyy-mm-dd hh24:mi:ss.ff'))         utc_ts
,      SYS_EXTRACT_UTC(to_timestamp_tz('2024-07-30 10:29:28.45678'       , 'yyyy-mm-dd hh24:mi:ss.ff'))         utc_ts_tz
,      SYS_EXTRACT_UTC(to_timestamp_tz('2024-07-30 10:29:28.45678 +01:00', 'yyyy-mm-dd hh24:mi:ss.ff TZH:TZM')) utc_ts_tz1
from   dual;
```
After executing it resulted in output with the UTC-variant stacked on top of the non-UTC version.
_For me that made comparing the related values a bit easier._
```
TS					 TS_TZ					  TS_TZ1
---------------------------------------- ---------------------------------------- ----------------------------------------
UTC_TS					 UTC_TS_TZ				  UTC_TS_TZ1
---------------------------------------- ---------------------------------------- ----------------------------------------
2024-07-30 10:29:28.456780000		 2024-07-30 10:29:28.456780000 +02:00	  2024-07-30 10:29:28.456780000 +01:00
2024-07-30 08:29:28.456780000		 2024-07-30 08:29:28.456780000		  2024-07-30 09:29:28.456780000
```
Hope this answered your question or convinced on your doubt.
_As you can see I'm in timezone: "UTC+02:00" as the column: "TS_TZ" automatically was appended with "+02:00"_

**Comment by okbob** [comment]
should be implemented in master, please, check

**Comment by okbob** [comment]
released in 4.11.0
